src/components/TxSubmitterResult.js

```diff
@@ -0,0 +1,181 @@
+              dispatch({ type: ACTIONS.submit });
+              const transaction = TransactionBuilder.fromXDR(
+                txXdr,
+                networkPassphrase,
+              );
+              const server = new Server(horizonURL, { appName: "Laboratory" });
+              server.submitTransaction(transaction).then(
+                (res) => {
+                  dispatch({ type: ACTIONS.success, payload: res });
+                },
+                (err) => {
+                  dispatch({ type: ACTIONS.fail, payload: err });
+                },
+              );
+            }}
+          >
+            {submission.state === NETWORK_STATES.pending
+              ? "Submitting to the network…"
+              : "Submit Transaction"}
+          </button>
+        </div>
+      </div>
+      <Collapsible isOpen={!isIdle}>
+        {!isIdle && submission.response && (
+          <Response {...submission.response} />
+        )}
+        {!isIdle && submission.error && <Error error={submission.error} />}
+      </Collapsible>
+    </div>
+  );
+};
+
+const Response = ({
+  operation_count,
+  successful,
+  paging_token,
+  hash,
+  ledger,
+  result_xdr,
+  result_meta_xdr,
+  fee_meta_xdr,
+}) => (
+  <div className="XdrViewer__submit so-back TransactionSubmitter__result">
+    <div className="so-chunk">
+      <h3 className="ResultTitle__success">Transaction submitted!</h3>
+      <p>
+        Transaction {successful ? "succeeded" : "failed"} with {operation_count}{" "}
+        operation(s).
+      </p>
+      <pre className="so-code TransactionSubmitter__code">
+        <code>
+          <div>
+            Hash:
+            <br /> <EasySelect>{hash}</EasySelect> <br />
+            Ledger number:
+            <br /> <EasySelect>{ledger}</EasySelect> <br />
+            Paging token:
+            <br /> <EasySelect>{paging_token}</EasySelect> <br />
+            Result XDR:
+            <br /> <EasySelect>{result_xdr}</EasySelect> <br />
+            Result Meta XDR:
+            <br /> <EasySelect>{result_meta_xdr}</EasySelect> <br />
+            Fee Meta XDR:
+            <br /> <EasySelect>{fee_meta_xdr}</EasySelect>
+          </div>
+        </code>
+      </pre>
+    </div>
+  </div>
+);
+const Error = ({ error }) => {
+  let message = "",
+    extras = null;
+  if (error instanceof AccountRequiresMemoError) {
+    message = "This destination requires a memo.";
+    extras = <React.Fragment>
+        Destination account:
+        <br /> <EasySelect>{error.accountId}</EasySelect> <br />
+        Operation index:
+        <br /> <EasySelect>{error.operationIndex}</EasySelect> <br />
+        </React.Fragment>
+  } else if (error instanceof BadResponseError) {
+    message = "An unknown error occurred";
+    extras = (
+      <React.Fragment>
+        original error:
+        <br />
+        <EasySelect>{JSON.stringify(error, null, 2)}</EasySelect>
+        <br />
+      </React.Fragment>
+    );
+  } else {
```

Comment:
btw, I think it would be worth checking that the type of error has `response` -- it might be the case that a different error appears and that would be the `else {` scenario

Reply:
Is there an error you have in mind that could cause that? I don't want to add stuff "just in case"—really, I'd love if the SDK could handle some of this complexity.

Reply:
@vcarl the SDK tries to handle as many errors as possible, however, since we rely on third parties too there might be errors that it doesn't know how to handle so it just bubbles it up to the consumer. 

In this particular scenario I don't think we are adding another handler just in case, but rather, handling unknown errors which might still happen, the else is assuming all other errors are of a given type, which isn't the case.

Reply:
Updated to make sure this will correctly handle any error thrown at it.